sdk/core/core-http/src/webResource.ts

```diff
@@ -108,7 +151,7 @@ export class WebResource {
     this.streamResponseBody = streamResponseBody;
     this.url = url || "";
     this.method = method || "GET";
-    this.headers = headers instanceof HttpHeaders ? headers : new HttpHeaders(headers);
+    this.headers = isHttpHeadersLike(headers) ? headers : new HttpHeaders(headers);
```

Comment:
I much prefer this way of checking the type of an object :)

Reply:
It's definitely not perfect, but yes better than relying on instanceof